modules/gapi/src/streaming/onevpl/cfg_params_parser.cpp

```diff
@@ -86,6 +96,76 @@ std::vector<ValueType> get_params_from_string(const std::string& str) {
+                                         false));
+        } else if (name == CfgParam::vpp_out_fourcc_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint32_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_chroma_format_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_width_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_height_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_crop_w_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_crop_h_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_crop_x_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_crop_y_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_pic_struct_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint16_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_framerate_n_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint32_t>(strtoul_or_throw(value.c_str())),
+                                         false));
+        } else if (name == CfgParam::vpp_out_framerate_d_name()) {
+            ret.push_back(creator.create(name,
+                                         static_cast<uint32_t>(strtoul_or_throw(value.c_str())),
+                                         false));
```

Comment:
It impossible to use switch for non-integral types: `string` is used here

Reply:
Oh, that's lame.. Sorry

Still, you can unite a lot of ifelse branches which do the exact same push_back

Reply:
Sorry, don't catch the idea:
Do you mean

```
if (name == A || name == B || name = C) 
{
push_back(name, **strtoull**);
} else if(name == D) 
{
push_back(name, **strtoll**);
} 
```
is it true?

Reply:
I think yes, you got me right

The idea is to combine all the conditions with the same action to one condition  
It will increase readability and shorten things